src/Widgets/TerminalWidget.vala

```diff
@@ -776,21 +776,27 @@ namespace Terminal {
                     var uris = selection_data.get_uris ();
                     string path;
                     File file;
-
                     for (var i = 0; i < uris.length; i++) {
-                         file = File.new_for_uri (uris[i]);
-                         if ((path = file.get_path ()) != null) {
-                             uris[i] = Shell.quote (path) + " ";
+                        // Get sanitized unquoted path (Files sends uris that are escaped and quoted)
+                        // We do not want the `file://` scheme included
+                        // We assume dropped paths are absolute
+                        file = File.new_for_uri (Utils.sanitize_path (Shell.unquote (uris[i]), "", false));
```

Comment:
This line causes the following warning on build:

```
../src/Widgets/TerminalWidget.vala:783.71-783.93: warning: unhandled error `GLib.ShellError'
  783 |                         file = File.new_for_uri (Utils.sanitize_path (Shell.unquote (uris[i]), "", false));
      |                                                                       ^~~~~~~~~~~~~~~~~~~~~~~              
```

Reply:
ShellError now handled